src/hooks/use-ordered-rows.ts

```diff
@@ -2,9 +2,12 @@ import { useMemo } from 'preact/hooks';
 
 import type { Order } from '../types';
 
+const collator = new Intl.Collator(undefined, { sensitivity: 'case' });
```

Comment:
Due to how we export modules in this library, this object will be created as soon as anything is imported from `@hypothesis/frontend-shared`.

We could implement some lazy-loading mechanism that ensures there's only one collator instance that's not created until the first time it's going to be used, but I didn't do it for simplicity.

```ts
let collator: Intl.Collator;

function getCollator(): Intl.Collator {
  if (!collator) {
    collator = new Intl.Collator(undefined, { sensitivity: 'case' });
  }

return collator;
}

// [...]

return getCollator().compare(x, y);
```

Reply:
I think this should be OK as this instantiation doesn't have global side effects. From my past experience collators are expensive enough that you don't want to create them eg. a sorting comparison function or on each render of a frequently used component (like an icon or something), but creating one or two globally should be fine.